File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT2_JAVA/src/Organisms/Animals/Turtle.java
package Organisms.Animals;

import Game.World;
import Organisms.Organism;

import java.awt.*;
import java.io.Serializable;

public class Turtle extends Animal implements Serializable {
    public Turtle(World world, int x, int y) {
        super("T", "Turtle", new Color(3,211,252), 2, 1, 1, world,x,y);
    }
    @Override
    public Turtle getChild(int x, int y)
    {
        return new Turtle(world,x,y);
    }
    @Override
    public boolean IsShielding(Organism attacker)
    {
        return attacker.getStrength()<5;
    }
    @Override
    public void Action()
    {
        if(world.getRandomNumber(4)==0)
        {
            super.Action();
        }
    }
}

File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT3_PYTHON/Organisms/Animals/turtle.py
import random

from Organisms.Animals.animal import Animal


class Turtle(Animal):
    def __init__(self, position_x, position_y, world):
        super().__init__(position_x, position_y, "Turtle", 'T', 2, 1, world, 1, (30, 253, 255))

    def getchild(self, newposition):
        return Turtle(newposition[0], newposition[1], self.world)

    def isshielding(self, attacker):
        if attacker.strength < 5:
            return True
        else:
            return False

    def action(self):
        i = random.randint(1,4)
        if i == 1:
            super().action()